cluster-autoscaler/estimator/estimation_context.go

```diff
@@ -0,0 +1,68 @@
+/*
+Copyright 2023 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package estimator
+
+import (
+	"k8s.io/autoscaler/cluster-autoscaler/cloudprovider"
+)
+
+// EstimationContext stores static and runtime state of autoscaling, used by Estimator
+type EstimationContext struct {
+	similarNodeGroups   []cloudprovider.NodeGroup
+	currentNodeCount    int
+	clusterMaxNodeLimit int
+}
+
+// NewEstimationContext creates new estimation context with static properties
+func NewEstimationContext(clusterMaxNodeLimit int) *EstimationContext {
+	return &EstimationContext{clusterMaxNodeLimit: clusterMaxNodeLimit}
+}
+
+// NewEstimationContextUpdate creates a patch for estimation context with runtime properties.
+// This patch is used to update existing context.
+func NewEstimationContextUpdate(similarNodeGroups []cloudprovider.NodeGroup, currentNodeCount int) *EstimationContext {
+	return &EstimationContext{
+		similarNodeGroups: similarNodeGroups,
+		currentNodeCount:  currentNodeCount,
+	}
+}
+
+// GetSimilarNodeGroups returns array of similar node groups
+func (c *EstimationContext) GetSimilarNodeGroups() []cloudprovider.NodeGroup {
```

Comment:
nit: getter

Reply:
Renamed in f3dfeeeb47a021ca4b854be5363e1a1663313b3b